lib/ai/process-chat-response.ts

```
@@ -0,0 +1,371 @@
+
+      const { value: partialArgs } = parsePartialJson(partialToolCall.text);
+
+      const invocation = {
+        state: 'partial-call',
+        step: partialToolCall.step,
+        toolCallId: value.toolCallId,
+        toolName: partialToolCall.toolName,
+        args: partialArgs,
+      } as const;
+
+      message.toolInvocations![partialToolCall.index] = invocation;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)